File: Introduction to Data Structures: Array and String/1. Introduction to Array/Plus One/PlusOne.java
class Solution {
    public int[] plusOne(int[] digits) {
        
        // edge case if length of input is 0
        if (digits.length == 0) {
            return digits;
        }
        
        // create a pointer to go right -> left
        int pointer = digits.length - 1;
        
        // create a counter to keep track of the 1's we need to carry
        int carry = 1;
        
        // while you have a 1 to carry and pointer > -1
        while (carry == 1 && pointer > -1) {
        
            // add one to the current element
            digits[pointer]++;
        
            // decrement the carry
            carry--;
        
            // if that element becomes a 10
            if (digits[pointer] == 10) {
        
                // change it to a 0
                digits[pointer] = 0;
        
                // increment the carry 
                carry++;
        
                // decrement the pointer 
                pointer--;
            }
        }
        // if the carry is not 0 (digits is some variation of [0,0,0,0,0])
        if (carry != 0) {
            // create a new array of an extra size
            digits = new int [1 + digits.length];

            // add 1 to the beggining of the array
            digits[0] = 1;
        }
        
        // return input array
        return digits;
        
                
    }
}

File: Introduction to Data Structures: Array and String/1. Introduction to Array/Plus One/PlusOne.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:

        # input : int array
        # output: int array

        # increment the large integer by one and return the resulting array of digits.

        # edge case: digits length is 0
        if len(digits) == 0:
            return nums

        # create a counter for 1
        oneCount = 1

        # create a pointer
        p = len(digits) - 1

        # iterate from the last index while counter == 1 and pointer > -1
        while oneCount == 1 and p > -1:

            # add 1 to the element
            digits[p] += 1

            # decrement one count
            oneCount -= 1

            # if the element becomes a 10
            if digits[p] == 10:

                # change element to 0
                digits[p] = 0

                # inc 1 to the counter
                oneCount += 1

                # decrement pointer
                p -= 1

        # insert 1 at 0 index if oneCount = 1
        if oneCount == 1:
            digits.insert(0, 1)

        # return digits
        return digits
